### PR TITLE
Enable -Xfatal-warning for all subprojects

### PR DESCRIPTION
**Pull request by itegulov**
## Overview
- Resolves all warnings in `models` and `blockstorage`
- Incorporates `almostCompilerSettings` into `compilerSettings`

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3432

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: models/src/main/scala/coop/rchain/models/ProtoM.scala [hunk 1]
```diff
@@ -228,7 +230,9 @@ object ProtoM extends DescriptorPimps {
         case FIXED32  => computeFixed32Size    (field.getNumber, value.asInstanceOf[Int])
         case BOOL     => computeBoolSize       (field.getNumber, value.asInstanceOf[Boolean])
         case STRING   => computeStringSize     (field.getNumber, value.asInstanceOf[String])
-        case GROUP    => computeGroupSize      (field.getNumber, value.asInstanceOf[MessageLite])
+        case GROUP    => throw new UnsupportedOperationException(
+          s"Groups are not deprecated, got $value of type ${value.getClass}"
```
Comment (ArturGajowy) [inline models/src/main/scala/coop/rchain/models/ProtoM.scala]: I now think that both `case`-es for `GROUP` (here and above) should `throw new UnsupportedOperationException("Groups are not supported, becuase they're deprecated in protobuf")

**Comment by ArturGajowy** [comment]
Thank you for doing this, BTW!

**Comment by itegulov** [comment]
@ArturGajowy PTAL

File: project/CompilerSettings.scala [hunk 2]
```diff
@@ -26,6 +26,7 @@ object CompilerSettings {
       "-feature",
       "-language:_",
       "-unchecked",
+      "-Xfatal-warnings",
```
Comment (rabbitonweb) [inline project/CompilerSettings.scala]: missing -unpacked

Comment (itegulov) [reply project/CompilerSettings.scala]: What is -unpacked? I can't find such an option in scalac.

**Pull request merged**